### ISSUE TITLE
If you remap OPTION+TAB to CMD+TAB, moving mouse while holding OPTION cancels the task selection

### ISSUE THREAD

**Issue by kbirger** [pqrs-org/Karabiner-Elements#2642]
I'm not sure how to describe this any better. I've created a rule like this:
```
{
                        "description": "alt tab",
                        "manipulators": [
                            {
                                "from": {
                                    "key_code": "tab",
                                    "modifiers": {
                                        "mandatory": [
                                            "option"
                                        ],
                                        "optional": [
                                            "any"
                                        ]
                                    }
                                },
                                "to": [
                                    {
                                        "key_code": "tab",
                                        "modifiers": "command"
                                    }
                                ],
                                "type": "basic"
                            }
                        ]
                    }
}
```
It is to allow me to use ALT+TAB (mapped to OPTION+TAB) on my Windows (us-101) keyboard instead of CMD+TAB. It works fairly well, but if press ALT+TAB and move my mouse, it cancels the window switcher. 

This same behavior happens for other multi-key combinations that involve holding down keys.

**Comment by terrywh** [comment]
+1

**Comment by kbirger** [comment]
It kind of seems like Karabiner conflates input devices because in the UI, it looks like some devices may report as having both keyboard and mouse functionality, which is fair enough (bluetooth combo keyboard/touchpad devices? Mice with Macros?), but it almost seems like mouse movement comes in through the same channel and cancels the key-down event

**Comment by aklingam7** [comment]
I have a Lenovo Bluetooth keyboard/mouse combo that shares a single dongle. 
The shortcut cancels when:
1. I move my external mouse while performing the shortcut on the external keyboard
2. I use the trackpad while performing the shortcut on my laptop keyboard
3. I use trackpad when performing the shortcut on the external keyboard
4. I move the mouse when performing the shortcut on my laptop keyboard.

Here's my rule:
```
      "description": "Ctrl + Tab to Cmd + Tab",
      "manipulators": [
        {
          "from": {
            "key_code": "tab",
            "modifiers": {
              "mandatory": [
                "left_control"
              ]
            }
          },
          "to": [
            {
              "key_code": "tab",
              "modifiers": [
                "left_gui"
              ]
            }
          ],
          "type": "basic"
        }
      ]
```

**Comment by kbirger** [comment]
It'd be just lovely to get some acknowledgement of this issue 🤔 . Perhaps if one of the maintainers knows the cause, it would encourage someone else to step up and fix it with a PR.

**Comment by Anachrion** [comment]
Same problem here.

**Comment by ksrb** [comment]
Having this issue as well, haven't tried it myself yet, but this thread seems to have some promise:
https://github.com/pqrs-org/Karabiner-Elements/issues/281

**Comment by bdalenoord** [comment]
I'm still affected by this issue, so the stale label is incorrect.

**Comment by franco-ruggeri** [comment]
@d-t-o have you found any workaround?

**Comment by d-t-o** [comment]
@franco-ruggeri yes, but it's most likely not the reply you want to hear: I use a DefaultKeyBinding.dict, a org.custom.keyboard-remap.plist (one for fixed rebinds (ctrl<->alt), one for combinations(end-key to end-of-line)), a self-defined keyboard layout made with ukelele (to enter such 'complicated' things as '~' or have my '@' on altgr+q) and AltTab from the app store to rebind my window switcher to alt-tab instead of ctrl-tab.

the only mapping I'm using Karabiner for, is remapping ctrl-tab to ctrl-shift-tab as ctrl-tab can't be rebound through other ways as it's reserved for the macos app switcher. since I don't need an app switcher on ctrl-tab if I have a window switcher on alt-tab, I had to free that combination up somehow. Doing this with Karabiner does not trigger the mouse-is-treated-as-input-bug we're talking about in here, but it's way less convenient than it could have been with doing everything in Karabiner.

**Comment by krykra7** [comment]
@d-t-o Hello, I'm newcomer to macos and I'm trying to achieve what you did, could you please give some resources about editing that dict files?

**Comment by adrian-v-popescu** [comment]
> Having this issue as well, haven't tried it myself yet, but this thread seems to have some promise #281.

One of the referenced workarounds is working for me. 

`{
                        "description": "Sticky Alt to Cmd Variable",
                        "manipulators": [
                          {
                            "type": "basic",
                            "from": {
                              "key_code": "left_alt",
                              "modifiers": {
                                "optional": ["any"]
                              }
                            },
                            "to": [
                              {
                                "set_variable": {
                                  "name": "left_alt pressed",
                                  "value": 1
                                }
                              },
                              {
                                "key_code": "left_command"
                              }
                            ],
                            "to_after_key_up": {
                              "set_variable": {
                                "name": "left_alt pressed",
                                "value": 0
                              }
                            }
                          }
                        ]
                    },

                    {
                        "description": "Alt+Tab to change app",
                        "manipulators": [
                            {
                                "conditions": [
                                    {
                                        "name": "left_alt pressed",
                                        "type": "variable_if",
                                        "value": 1
                                    }
                                ],
                                "from": {
                                    "key_code": "tab",
                                    "modifiers": {
                                        "mandatory": ["option"],
                                        "optional": ["any"]
                                    }
                                },
                                "to": [
                                    {
                                        "key_code": "tab",
                                        "modifiers": [
                                            "left_command"
                                        ]
                                    }
                                ],
                                "type": "basic"
                            }
                        ]
                    }`

**Comment by kbirger** [comment]
This works, but it also overrides other `ALT + *` sequences, like `ALT + LEFT` which is supposed to move the caret back a word, and `ALT + delete` which is supposed to remove a word.

**Comment by adiletelf** [comment]
Same issue here. Mapping `Cmd+Tab` to `Option+Tab` works ok, but then if I move the mouse or press other modifiers, the mapping stops working. Example: Press `Cmd+Tab` to trigger switching application using [AltTab](https://alt-tab-macos.netlify.app/) shortcut `Option+Tab`. Works great. But if I press `Shift` to switch to previous application, the complex_modification stops working. Same if I move my mouse.

I tried to work around it by adding the rule `Cmd+LeftShift` -> `Option+LeftShift` to allow switching to previous app and it works, but not only in "AltTab" application switch popup but also in all other applications. I tried to add `frontmost_application_if` with "AltTab" bundle_identifier `com.lwouis.alt-tab-macos` but it only works for actual "AltTab" app but not for popup. Karabiner-EventViewer didn't output the bundle_identifier for AltTab popup so I'm stuck.

My code:
```json
{
  "description": "Map <Cmd-Tab> to <M-Tab> for AltTab app",
  "manipulators": [
    {
      "type": "basic",
      "from": {
        "key_code": "tab",
        "modifiers": {
          "mandatory": ["left_command"],
          "optional": ["any"]
        }
      },
      "to": [
        {
          "key_code": "tab",
          "modifiers": ["left_option"]
        }
      ]
    },
    {
      "type": "basic",
      "conditions": [
        {
          "bundle_identifiers": ["com.lwouis.alt-tab-macos"],
          "file_paths": ["/Applications/AltTab.app/Contents/MacOS/AltTab"],
          "type": "frontmost_application_if"
        }
      ],
      "from": {
        "key_code": "left_shift",
        "modifiers": {
          "mandatory": ["left_command"]
        }
      },
      "to": [
        {
          "key_code": "left_shift",
          "modifiers": ["left_option"]
        }
      ]
    }
  ]
}
```

If there is a way to find out the bundle_identifier for "AltTab" popup switcher, then my issue will be solved (though moving mouse still would mess everything).

**Comment by hgyxbll** [comment]
Meet Same issue. 
I Mapping Cmd+Tab to Ctrl+Tab , it  works ok, but then if I move the mouse, the mapping stops working.
use EventViewer to check what happened, below is log:

```json
[
  {
    "type": "down",
    "name": {"key_code":"left_command"},
    "usagePage": "7 (0x0007)",
    "usage": "227 (0x00e3)",
    "misc": "flags left_command"
  },
  {
    "type": "down",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_command"
  },
//press tab, it start  map cmd as ctrl
  {
    "type": "down",
    "name": {"key_code":"left_control"},
    "usagePage": "7 (0x0007)",
    "usage": "224 (0x00e0)",
    "misc": "flags left_command,left_control"
  },
  {
    "type": "up",
    "name": {"key_code":"left_command"},
    "usagePage": "7 (0x0007)",
    "usage": "227 (0x00e3)",
    "misc": "flags left_control"
  },
  {
    "type": "down",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_control"
  },
  {
    "type": "up",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_option"
  },
  {
    "type": "up",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_control"
  },
  {
    "type": "down",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_option"
  },
  {
    "type": "down",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_control"
  },
  {
    "type": "up",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_option"
  },
  {
    "type": "up",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_control"
  },
// move mouse, it release ctrl and press cmd
  {
    "type": "up",
    "name": {"key_code":"left_control"},
    "usagePage": "7 (0x0007)",
    "usage": "224 (0x00e0)",
    "misc": ""
  },
  {
    "type": "down",
    "name": {"key_code":"left_command"},
    "usagePage": "7 (0x0007)",
    "usage": "227 (0x00e3)",
    "misc": "flags left_command"
  },
  {
    "type": "down",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_option"
  },
//press tab again, it map cmd as ctrl again.
  {
    "type": "down",
    "name": {"key_code":"left_control"},
    "usagePage": "7 (0x0007)",
    "usage": "224 (0x00e0)",
    "misc": "flags left_command,left_control"
  },
  {
    "type": "up",
    "name": {"key_code":"left_command"},
    "usagePage": "7 (0x0007)",
    "usage": "227 (0x00e3)",
    "misc": "flags left_control"
  },
  {
    "type": "down",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_control"
  },
  {
    "type": "up",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_option"
  },
  {
    "type": "up",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_control"
  },
  {
    "type": "down",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_option"
  },
  {
    "type": "down",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_control"
  },
  {
    "type": "up",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_option"
  },
  {
    "type": "up",
    "name": {"key_code":"tab"},
    "usagePage": "7 (0x0007)",
    "usage": "43 (0x002b)",
    "misc": "flags left_control"
  },
  {
    "type": "up",
    "name": {"key_code":"left_option"},
    "usagePage": "7 (0x0007)",
    "usage": "226 (0x00e2)",
    "misc": ""
  },
  {
    "type": "up",
    "name": {"key_code":"left_control"},
    "usagePage": "7 (0x0007)",
    "usage": "224 (0x00e0)",
    "misc": ""
  },

]
```
do not know why move mouse will press cmd again.

**Comment by DvoeglazovIlya** [comment]
Maybe it help, sorry for my English.
I use windows keyboard, therefore I have on the left side "Ctrl", "Command" (or "Win"), "Option" (or "Alt").
I think this trick work only with app where you can change shortcuts.

I want copy/paste like in windows therefore I switch "Ctrl" to "Command" and "Command" to "Ctrl" in settings "Simple Modification" and I add another key to this settings "f16" to "f16" (I don't have this key in my keyboard, but I explain it later, you can add unused key or add not existing key like I). 

![image](https://github.com/pqrs-org/Karabiner-Elements/assets/15049055/f4e043b9-7d45-4135-9dae-9759097780fa)

Now copy/paste work like in windows, but switching tabs in browser like in windows not working because "Ctrl" now is "Cmd" that is incorrect position in keyboard. In windows I can switch tabs in browser like this "Ctrl+(Shift)+Tab", therefore in macOs I need to change it like this "Cmd+(Shift)+Tab". See code below:

```json
{
    "description": "Cmd+(Shift)+Tab => Ctrl+(Shift)+Tab (Switch tabs) (Add condition for app where you have problem add custom shortcuts in settings (for example see VSCode Cmd+F16))",
    "manipulators": [{
            "conditions": [{
                "bundle_identifiers": [
                    "^com\\.microsoft\\.VSCode"
                ],
                "type": "frontmost_application_unless"
            }],
            "from": {
                "key_code": "tab",
                "modifiers": {
                    "mandatory": [
                        "command"
                    ],
                    "optional": [
                        "shift",
                        "caps_lock"
                    ]
                }
            },
            "to": [{
                "key_code": "tab",
                "modifiers": [
                    "control"
                ]
            }],
            "type": "basic"
        },
        {
            "from": {
                "key_code": "tab",
                "modifiers": {
                    "mandatory": [
                        "command"
                    ],
                    "optional": [
                        "shift",
                        "caps_lock"
                    ]
                }
            },
            "to": [{
                "key_code": "f16",
                "modifiers": [
                    "command"
                ]
            }],
            "type": "basic"
        }
    ]
}
``` 

You can see the condition for VSCode and f16 key. f16 need in VSC to change shortcuts. 
Because when using "Cmd+(Shift)+Tab" and moving mouse while holding "Cmd" or pressing "Shift" cancels the task selection (like in above comments) therefore I add not existing key and change VSC shortcuts to switch tabs. 
So I modify combination "Cmd+tab" to "Cmd+f16" only in VSC. The order in "manipulators" elements is required.
Find and change shortcut "ctrl+tab" and press "cmd+tab" it writes "cmd+f16".
Find "ctrl+shift+tab" and press "cmd+shift+tab" it writes "shift+cmd+f16".

![image](https://github.com/pqrs-org/Karabiner-Elements/assets/15049055/3d591551-1451-4090-8413-ac6dec8c7358)

Or just add shortcuts direct to keybindings.json.

My keybindings.json looks:
```json
[
    {
        "key": "cmd+f16",
        "command": "workbench.action.quickOpenPreviousRecentlyUsedEditorInGroup",
        "when": "!activeEditorGroupEmpty"
    },
    {
        "key": "ctrl+tab",
        "command": "-workbench.action.quickOpenPreviousRecentlyUsedEditorInGroup",
        "when": "!activeEditorGroupEmpty"
    },
    {
        "key": "cmd+f16",
        "command": "workbench.action.quickOpenNavigateNextInEditorPicker",
        "when": "inEditorsPicker && inQuickOpen"
    },
    {
        "key": "ctrl+tab",
        "command": "-workbench.action.quickOpenNavigateNextInEditorPicker",
        "when": "inEditorsPicker && inQuickOpen"
    },
    {
        "key": "shift+cmd+f16",
        "command": "workbench.action.quickOpenLeastRecentlyUsedEditorInGroup",
        "when": "!activeEditorGroupEmpty"
    },
    {
        "key": "ctrl+shift+tab",
        "command": "-workbench.action.quickOpenLeastRecentlyUsedEditorInGroup",
        "when": "!activeEditorGroupEmpty"
    },
    {
        "key": "shift+cmd+f16",
        "command": "workbench.action.quickOpenNavigatePreviousInEditorPicker",
        "when": "inEditorsPicker && inQuickOpen"
    },
    {
        "key": "ctrl+shift+tab",
        "command": "-workbench.action.quickOpenNavigatePreviousInEditorPicker",
        "when": "inEditorsPicker && inQuickOpen"
    }
]
``` 
Don't forgot about "conditions" in code above. For VSC I add "^com\\.microsoft\\.VSCode". This condition means use "Cmd+Tab" everywhere except VSC, and last element in "manipulators" intended for apps (in my case VSC) where you have problem and where you can change shortcuts.

**Comment by fatihacarli** [comment]
I am also having this problem with Karabiner 14.13.0 and OSX 13.6

**Comment by jameshbailie** [comment]
> Thanks to vandemjh's comments on #281 I finally have this working properly. Recommend you read his comments for explanation, but basically we use two rules.

This works for me!

**Comment by ngrigoriev** [comment]
+1. And it also affects other events when you move the mouse at the same time when clicking the key combination that is subject to modifications.

I often move the mouse while switching tasks because I remember which corner of the screen the app is in, this is how I stumbled myself on this issue. Glad not to be alone :)

**Comment by michael-vostrikov** [comment]
I made settings to make keys work similar to Ubuntu/Windows, including Alt+Tab without problem with mouse.

Place attached file to following location:
```
Users/[user]/.config/karabiner/assets/complex_modifications
```

Open "Karabiner - Simple Modifications"

Add modifications:
```
fn -> left_control
left_control -> fn
```

Open "Complex Modifications"

Add rules from "Normal keys" section as needed. Order of rules matters.

There is a special rule "CapsLock+DigitsLine to F1..F12", it is done for debugging. Digits work as F1..F12 when CapsLock is on. You need to switch off CapsLock if you want to write something during debugging, then switch it on back. It may get annoying, so you can remove this rule.

[166273300.json](https://github.com/user-attachments/files/15525125/166273300.json)

**Comment by SteliosKornelakis** [comment]
> I made settings to make keys work similar to Ubuntu/Windows, including Alt+Tab without problem with mouse.
> 
> Place attached file to following location:
> 
> ```
> Users/[user]/.config/karabiner/assets/complex_modifications
> ```
> 
> Open "Karabiner - Simple Modifications"
> 
> Add modifications:
> 
> ```
> fn -> left_control
> left_control -> fn
> ```
> 
> Open "Complex Modifications"
> 
> Add rules from "Normal keys" section as needed. Order of rules matters.
> 
> There is a special rule "CapsLock+DigitsLine to F1..F12", it is done for debugging. Digits work as F1..F12 when CapsLock is on. You need to switch off CapsLock if you want to write something during debugging, then switch it on back. It may get annoying, so you can remove this rule.
> 
> [166273300.json](https://github.com/user-attachments/files/15525125/166273300.json)

Such a great response, finally working around this annoying mouse move issue!
A big Thank You, @michael-vostrikov !

**Comment by franco-ruggeri** [comment]
> > This bug is really annoying, so while I can't offer much I'd be willing to pay a bounty of 30€ or 30USD to the person fixing this bug in an official release of Karabiner Elements.
> 
> Thanks to vandemjh's comments on #281 I finally have this working properly. Recommend you read his comments for explanation, but basically we use two rules.
> 
> The first rule "convert[s] the `left_alt` press into a `left_command` press with the side effect of setting a variable, `left_alt pressed` to 1 while `left_alt` is being pressed (and zero otherwise)" (which lets us use alt-tab without removing the ability to use alt in other keycombos). So, the first rule:
> 
> ```json
> {
>     "description": "Sticky Alt to Cmd Variable",
>     "manipulators": [
>         {
>             "from": {
>                 "key_code": "left_alt",
>                 "modifiers": {
>                     "optional": [
>                         "any"
>                     ]
>                 }
>             },
>             "to": [
>                 {
>                     "set_variable": {
>                         "name": "left_alt pressed",
>                         "value": 1
>                     }
>                 },
>                 {
>                     "key_code": "left_command"
>                 }
>             ],
>             "to_after_key_up": {
>                 "set_variable": {
>                     "name": "left_alt pressed",
>                     "value": 0
>                 }
>             },
>             "type": "basic"
>         }
>     ]
> }
> ```
> 
> Then we can create a simple Ctrl+Tab => Cmd+Tab rule that takes the `left_alt pressed` variable and uses it as a condition:
> 
> ```json
> {
>     "description": "Ctrl+Tab => Cmd+Tab (Switch application) using Sticky Alt to Cmd Variable",
>     "manipulators": [
>         {
>             "conditions": [
>                 {
>                     "name": "left_alt pressed",
>                     "type": "variable_if",
>                     "value": 1
>                 }
>             ],
>             "from": {
>                 "key_code": "tab",
>                 "modifiers": {
>                     "mandatory": [
>                         "command"
>                     ],
>                     "optional": [
>                         "any"
>                     ]
>                 }
>             },
>             "to": [
>                 {
>                     "key_code": "tab",
>                     "modifiers": [
>                         "left_command"
>                     ]
>                 }
>             ],
>             "type": "basic"
>         }
>     ]
> }
> ```

+1
It works for me too.

**Comment by michael-vostrikov** [comment]
"cmd+shift+3, cmd+shift+4" become "fn+shift+3, fn+shift+4", they work.
I see "cmd+ctrl+shift+3, cmd+ctrl+shift+4" do not work, I don't use them, so I did not notice. I think special rule for them in a separate file added before others should help.

**Comment by feliperdamaceno** [comment]
I've modified the version from @d-allin abit, just to also disable the default cmd+tab so that I can use that keymap in other scenarios, just in case someone is looking for it :D

```JSON
{
    "description": "Sticky Alt to Cmd Variable and Custom Ctrl+Tab => Cmd+Tab with Cmd+Tab Disabled",
    "manipulators": [
        {
            "from": {
                "key_code": "left_alt",
                "modifiers": { "optional": ["any"] }
            },
            "to": [
                {
                    "set_variable": {
                        "name": "left_alt pressed",
                        "value": 1
                    }
                },
                { "key_code": "left_command" }
            ],
            "to_after_key_up": {
                "set_variable": {
                    "name": "left_alt pressed",
                    "value": 0
                }
            },
            "type": "basic"
        },
        {
            "conditions": [
                {
                    "name": "left_alt pressed",
                    "type": "variable_if",
                    "value": 1
                }
            ],
            "from": {
                "key_code": "tab",
                "modifiers": {
                    "mandatory": ["command"],
                    "optional": ["any"]
                }
            },
            "to": [
                {
                    "key_code": "tab",
                    "modifiers": ["left_command"]
                }
            ],
            "type": "basic"
        },
        {
            "description": "Disable default Cmd+Tab",
            "from": {
                "key_code": "tab",
                "modifiers": {
                    "mandatory": ["command"],
                    "optional": ["any"]
                }
            },
            "to": [{ "key_code": "vk_none" }],
            "type": "basic"
        }
    ]
}
```

**Comment by Eloren1** [comment]
↑ best solution, thx

**Comment by arthurdorigueto** [comment]
I did the following to solve the problem of having "the same" keyboard layout both on Windows and MacOS. 

1. On Karabiner add two Simple Modifications

left_command -> left_control
left_control -> left command

This already solves 98% of all shortcuts that I can think of.

2. On Karabiner add a Complex Modification

The modification changes Cmd+Tab to any other shortcut. In my case, I used the following: 

```
{
    "description": "Cmd+Tab to Cmd+F13 (optional Shift) for Chrome Tab Switching",
    "manipulators": [
        {
            "from": {
                "key_code": "tab",
                "modifiers": {
                    "mandatory": ["left_command"],
                    "optional": ["shift"]
                }
            },
            "to": [
                {
                    "key_code": "f13",
                    "modifiers": ["left_command"]
                }
            ],
            "type": "basic"
        }
    ]
}
```

The new shortcut should be complicated so it does not conflict with other known shortcuts. 

3. On Keyboard -> Shortcuts -> App Shortcuts, I add a custom shortcut for my browser (chrome) to change tabs using the shortcut defined in the previous step. 

The idea here is that when I press Cmd+Tab I will change tabs without triggering the default Cmd+Tab.

4. Finally, I use AltTab (app) to set the shortcut for task selection to Control+Tab, ending the setup.

**Comment by christophe-calmejane** [comment]
> > This bug is really annoying, so while I can't offer much I'd be willing to pay a bounty of 30€ or 30USD to the person fixing this bug in an official release of Karabiner Elements.
> 
> Thanks to vandemjh's comments on [#281](https://github.com/pqrs-org/Karabiner-Elements/issues/281) I finally have this working properly. Recommend you read his comments for explanation, but basically we use two rules.

I tried all the proposed solutions, but none is properly working (using variable, or direct complex manipulation).

Yes we can swap CMD/CTRL/ALT-TAB but each time, when the mouse is moved or if shift is pressed during the key sequence, the combo is stopped (whether its the app switcher or the tab switcher in Chrome or VSCode).

I don't think we'll have a proper solution until the mouse/other modifiers don't break the sequence.

**Comment by ad08fee3** [comment]
I found a viable setup for this! It works for both my external keyboard and the built-in keyboard, and moving the mouse doesn't break the AltTab app.

The following setup lets me:
* Press the key labeled `Ctrl` and `Tab` to change tabs in browsers, like Chrome.
* Press the key labeled `Alt` and `Tab` to activate application switching with alt-tab
* Press the key labeled `Windows` and `Tab` to activate mission control, similar to `Windows + Tab` in Windows 10/11.

It also works with the built-in Mac keyboard, if you are okay with the muscle-memory of key locations instead of the words printed on the keys.

---

Within AltTab I have the trigger set up to be Option + Tab:

<img width="548" height="224" alt="Image" src="https://github.com/user-attachments/assets/9805c90f-44c9-4097-80e0-9bbe3b32375a" />


Within system settings, I have this abomination of a shortcut set to activate Mission Control: (You'll see why.)

<img width="438" height="118" alt="Image" src="https://github.com/user-attachments/assets/9b730881-35f9-4ede-a5e3-6c3f967ebd50" />

My external keyboard has these simple modifications:
_(Essentially I swapped command and control.)_
```json
{
    "description": "My regular PC keyboard, plugged into a Mac",
    "identifiers": {
        "is_keyboard": true,
        "product_id": <whatever is your keyboard>,
        "vendor_id": <whatever is your keyboard>
    },
    "simple_modifications": [
        {
            "from": { "key_code": "left_command" },
            "to": [{ "key_code": "left_control" }]
        },
        {
            "from": { "key_code": "left_control" },
            "to": [{ "key_code": "left_command" }]
        },
        {
            "from": { "key_code": "right_control" },
            "to": [{ "key_code": "right_command" }]
        }
    ]
},
```

Then I have these complex modifications:
```json
{
    "description": "Control-Tab maps to some weird arbitrary combo to activate Mission Control.",
    "manipulators": [
        {
            "description": "The button labeled Windows is already mapped to Control. Thus, pressing Win+Tab gets mapped to Control+Tab. But to prevent this from cycling tabs in browsers (like Chrome/Firefox), we should map Control+Tab to some arbitrary key combo. It doesn't have to be this one; but whatever you use you should then set as the Mission Control keyboard shortcut in system settings.",
            "from": {
                "key_code": "tab",
                "modifiers": { "mandatory": ["control"] }
            },
            "to": [
                {
                    "key_code": "j",
                    "modifiers": ["control", "shift", "command", "option"]
                }
            ],
            "type": "basic"
        }
    ]
},
{
    "description": "Command-Tab becomes control-tab (for cycling through browsers)",
    "manipulators": [
        {
            "description": "The Control button on the keyboard is mapped to Command, so we're essentially re-mapping it BACK to control again when you press the tab button.",
            "from": {
                "key_code": "tab",
                "modifiers": {
                    "mandatory": ["command"],
                    "optional": ["shift"]
                }
            },
            "to": [
                {
                    "key_code": "tab",
                    "modifiers": ["control"]
                }
            ],
            "type": "basic"
        }
    ]
}
```

As for my built-in Mac keyboard, I have these simple modifications:
```json
"simple_modifications": [
    {
        "from": { "apple_vendor_top_case_key_code": "keyboard_fn" },
        "to": [{ "key_code": "left_command" }]
    },
    {
        "from": { "key_code": "left_command" },
        "to": [{ "key_code": "left_option" }]
    },
    {
        "from": { "key_code": "left_control" },
        "to": [{ "apple_vendor_top_case_key_code": "keyboard_fn" }]
    },
    {
        "from": { "key_code": "left_option" },
        "to": [{ "key_code": "left_control" }]
    },
    {
        "from": { "key_code": "right_command" },
        "to": [{ "key_code": "right_option" }]
    },
    {
        "from": { "key_code": "right_option" },
        "to": [{ "key_code": "right_command" }]
    }
]
```
This makes the `fn` key act as Control, the Command Key act as Alt, the Option Key act as Windows, and the Control key act as fn. That layout _physically_ matches most other Windows laptops that I've used, so that from left to right the keys are `Ctrl Fn Win Alt` instead of Mac's `Fn Ctrl Opt Cmd`.
I'm thinking of putting tape over the keyboard labels, actually.

But I can confirm that this setup works on the native mac keyboard too! 
You just need to mentally assume the key to the left of the space bar (labeled Command) is actually the Alt key.

TLDR: If you make the key left of the spacebar "Option" instead of "Command" then you can use AltTab with the shortcut Option-Tab and the mouse won't break any remapping :)

**Comment by arthurdorigueto** [comment]
@ad08fee3 that worked pretty well on the Windows keyboard.

**Comment by tekezo** [comment]
By using `to_if_other_key_pressed`, which was introduced in the latest beta (v15.9.17), you can now reliably open the application switcher with option+tab.
You can achieve this by using the latest beta along with the configuration on this page.
https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/to-if-other-key-pressed/

How to upgrade Karabiner-Elements to the latest beta version:
<https://karabiner-elements.pqrs.org/docs/manual/operation/check-for-updates/#using-the-latest-beta-version>

**Comment by christophe-calmejane** [comment]
I confirm it solved all the mouse events issues (Task switcher, visual code window switcher, ...). Thank you so much for that new feature!!
